### ISSUE TITLE
stuck at loading

### ISSUE THREAD

**Issue by GorvGoyl** [aarondill/Vercel-Project-Manager#1]
I provided the Vercel token and reloaded the VSCode window but the extension is stuck at loading. These are the errors I got:

<img width="501" alt="image" src="https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/5032c2df-1070-4902-b891-b75662c59fb6">

**Comment by aarondill** [comment]
Interesting. That is an error returned from the Vercel servers. Are you certain that you set the authentication token *correctly*? If so, can you provide a bit of context so I can attempt to debug? The only places `map` is called are when managing the environment list (which *shouldnt* ever be undefined) and when getting the full list of deployments. What is your `DeploymentCount` set to (or is it not, it has a default)? Do you have any deployments already?

If you open the bottom panel and click on `output`, there should be a drop down at the top right of the panel. Can you set it to this extension and provide the logs/error trace back that should be there?

**Comment by GorvGoyl** [comment]
> Are you certain that you set the authentication token correctly?

yes because same token is used for another vscode extension and its working properly (https://marketplace.visualstudio.com/items?itemName=HaydenBleasel.vercel-vscode)

DeploymentCount is 20

settings:

![screenshot 2023-07-22 at 17 52 38@2x](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/63104b5d-b19b-4416-9a80-2294294effc9)


> Can you set it to this extension and provide the logs/error trace back that should be there?
this extension is not available in that dropdown but some other extensions are.

**Comment by GorvGoyl** [comment]
also, I have one pro plan (team) and one hobby plan (personal) of Vercel i(f that helps in any way).

**Comment by aarondill** [comment]
@GorvGoyl are you still experiencing both of these errors?

**Comment by GorvGoyl** [comment]
unfortunately, yes.

**Comment by AppreciateIT** [comment]
I'm getting this error when I try to link to Vercel using VSCode in Windows 10.  Any clues or fixes would be appreciated:

(trap "exit \$?" INT && clear && echo "Type cmd/ctrl/meta + c to quit" && echo ; if ! command -v vercel &>/dev/null ; then echo "The vercel CLI is required for this feature.
>> Please install with npm install -g vercel" ; while true; do : ; done ; exit 1 ; fi ; vercel link -t ZE2UIJJplGAZOxW7UkLdaR2P ); ret=$? ; test $ret -eq 130 && exit 0 ; test $ret -eq 0 && exit $ret ; test $ret -nq 0 && exit $ret
ParserError: 
Line |
   1 |  … INT && clear && echo "Type cmd/ctrl/meta + c to quit" && echo ; if !  …
     |                                                                 ~
     | Missing closing ')' in expression.
PS C:\Users\mrmat> export PATH=$PWD/node_modules/.bin:$PATH
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

**Comment by aarondill** [comment]
@GorvGoyl ive recently updated the extension to login with OAuth. Could you update your extension to version 2.0.0 and see if this issue is still present?

**Comment by GorvGoyl** [comment]
I just tried and it's still not working. I was able to give OAuth but when I returned to the extension i got the error: Not Authorised. I tried reinstalling as well but it didn't work. 


![screenshot 2024-02-09 at 14 10 31@2x](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/fbec4495-99cd-408e-ad9c-4cc504a61484)

**Comment by GorvGoyl** [comment]
![screenshot 2024-02-09 at 14 25 41@2x](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/7bdd5940-c641-485e-a677-9afc8c5d992e)

**Comment by aarondill** [comment]
if you update the extension once more, do you still get the not authorized error on the latest version?

**Comment by GorvGoyl** [comment]
yes 
![screenshot 2024-02-10 at 17 26 30@2x](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/7f680f7e-ed51-494c-8850-3b156c0f4593)

**Comment by aarondill** [comment]
> also, I have one pro plan (team) and one hobby plan (personal) of Vercel i(f that helps in any way).

is the project you're testing with registered under the team plan or the personal plan?

**Comment by GorvGoyl** [comment]
under team plan.

**Comment by aarondill** [comment]
That explains it then!
According to the [vercel docs](https://vercel.com/docs/rest-api/vercel-api-integrations#using-the-vercel-api/scopes/teams)

```
403 Forbidden responses
Ensure you are not missing the teamId query parameter. teamId is required if the integration installation is for a Team. Ensure the Scope of Your Access Token is properly set.
```

now that I know what the problem is, hopefully I can solve it!

sadly, it does seem like the way their apis work, you'll only be able to use the extension for one plan at a time, so *either* team *or* hobby; unless you log out and back in between uses.

**Comment by aarondill** [comment]
the api responds with the team_id (or null) when returning the api token, so I just need to store it and apply it to all api requests

**Comment by aarondill** [comment]
is this fixed after #5?

**Comment by GorvGoyl** [comment]
still facing the issue in 2.1.0
![image](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/b9733af2-edea-4697-8201-9ca911ad413a)

**Comment by aarondill** [comment]
@GorvGoyl did you re-login after the update? if not, would you please try that? The old version didn't retain the teamId parameter, so in order to use it in 2.1.0 it must be regenerated and stored

**Comment by aarondill** [comment]
There was a regression introduced in 2.1.0 that made all attempts at authentication fail. I have just pushed 2.1.1 with the fix, please logging out and then logging in after updating to 2.1.1

**Comment by GorvGoyl** [comment]
from where do I log out?

**Comment by aarondill** [comment]
There is a command to do so: `vercel.logOut`. It's title is `Vercel Project Manager: Log Out`. You can run it through the command palette.

**Comment by aarondill** [comment]
@GorvGoyl did that fix it?

**Comment by GorvGoyl** [comment]
Thanks, it finally worked.
One feedback: I see the dates of deployments but it would be nice if I can get a little bit more info like `preview` vs `production` or deployed branch name
![screenshot 2024-02-20 at 18 35 27@2x](https://github.com/aarondill/Vercel-Project-Manager/assets/7106086/6de9b29b-e6f2-46ca-9028-352df17ddcd1)

**Comment by aarondill** [comment]
@GorvGoyl Im going to close this issue since it's been resolved. Feel free to open a new issue with your feature request described in detail 😃
